### PR TITLE
added value type option(s) support, added class RegexType

### PR DESCRIPTION
**Pull request by 1Franck**
```php
$specs->add('r|regex:', 'with custom regex type value')
      ->isa('Regex', '/^([a-z]+)$/');
```

### CODE REVIEW

**Comment by 1Franck** [comment]
did you see this PR? Is it ok with you?

**Comment by c9s** [comment]
ah, I missed the mail. looks good to me

**Comment by c9s** [comment]
Hi  @1Franck this PR is released in 2.2.1.

File: tests/GetOptionKit/ValueTypeTest.php [hunk 1]
```diff
@@ -74,5 +76,17 @@ public function testEmailType()
         ok( $email->test('test@gmail.com') );
         $this->assertFalse($email->test('test@test'));
     }
+
+    public function testRegexType()
+    {
+        $regex = new RegexType;
+        $regex->option = '#^Test$#';
```
Comment (c9s) [inline tests/GetOptionKit/ValueTypeTest.php]: Can we pass the regular expression to the constructor?

File: tests/GetOptionKit/ValueTypeTest.php [hunk 2]
```diff
@@ -74,5 +76,17 @@ public function testEmailType()
...
+        ok( $regex->test('Test') );
+        $this->assertFalse($regex->test('test'));
+
+        $regex->option = '/^([a-z]+)$/';
```
Comment (c9s) [inline tests/GetOptionKit/ValueTypeTest.php]: This should be another test case

File: tests/GetOptionKit/ValueTypeTest.php (hunk 1 above)
Comment (1Franck) [inline tests/GetOptionKit/ValueTypeTest.php]: Yes we can.

File: tests/GetOptionKit/ValueTypeTest.php (hunk 2 above)
Comment (1Franck) [inline tests/GetOptionKit/ValueTypeTest.php]: I can create a new test file called RegexValueTypeTest.php and move stuff there if its what you mean.

Comment (c9s) [inline tests/GetOptionKit/ValueTypeTest.php]: That would be great :)

**Pull request merged**